test/integration/roomMembersTest.ts

```diff
@@ -374,4 +377,200 @@ describe("Test: Testing RoomMemberManager", function() {
+                method: Method.kick,
+            },
+            // Ban bad users in one room, using duration syntax, no reason.
+            {
+                name: "ban with duration",
+                command: roomId => `!mjolnir since ${Date.now() - cutDate.getTime()}ms ban 100 ${roomId}`,
+                shouldAffectWitnessRoom: false,
+                n: 1,
+                method: Method.ban,
+            },
+            // Kick bad users in one room, using date syntax, no reason.
+            {
+                name: "kick with date",
+                command: roomId => `!mjolnir since "${cutDate}" kick 100 ${roomId}`,
+                shouldAffectWitnessRoom: false,
+                n: 1,
+                method: Method.kick,
+            },            
+            // Ban bad users in one room, using date syntax, no reason.
+            {
+                name: "ban with date",
+                command: roomId => `!mjolnir since "${cutDate}" ban 100 ${roomId}`,
+                shouldAffectWitnessRoom: false,
+                n: 1,
+                method: Method.ban,
+            },
+
+            // Kick bad users in one room, using duration syntax, with reason.
+            {
+                name: "kick with duration and reason",
+                command: roomId => `!mjolnir since ${Date.now() - cutDate.getTime()}ms kick 100 "bad, bad user" ${roomId}`,
+                shouldAffectWitnessRoom: false,
+                n: 1,
+                method: Method.kick,
+            },
+            // Ban bad users in one room, using duration syntax, with reason.
+            {
+                name: "ban with duration and reason",
+                command: roomId => `!mjolnir since ${Date.now() - cutDate.getTime()}ms ban 100 "bad, bad user" ${roomId}`,
+                shouldAffectWitnessRoom: false,
+                n: 1,
+                method: Method.ban,
+            },
+            // Kick bad users in one room, using date syntax, with reason.
+            {
+                name: "kick with date and reason",
+                command: roomId => `!mjolnir since "${cutDate}" kick 100 "bad, bad user" ${roomId}`,
+                shouldAffectWitnessRoom: false,
+                n: 1,
+                method: Method.kick,
+            },            
+            // Ban bad users in one room, using date syntax, with reason.
+            {
+                name: "ban with date and reason",
+                command: roomId => `!mjolnir since "${cutDate}" ban 100 "bad, bad user" ${roomId}`,
+                shouldAffectWitnessRoom: false,
+                n: 1,
+                method: Method.ban,
+            },
+            
+            // Kick bad users everywhere, no reason
+            {
+                name: "kick with date everywhere",
+                command: () => `!mjolnir since "${cutDate}" kick 100 "bad, bad user" *`,
+                shouldAffectWitnessRoom: true,
+                n: NUMBER_OF_ROOMS - 1,
```

Comment:
so for each room it sends a reply to the command and we are checking for the last reply to know whether the command succeeded?

Reply:
We use it to wait until the command has finished. The rest of the loop is checking whether the command actually succeeded.

Reply:
I've added a little documentation.